share/functions/__fish_git_prompt.fish

```diff
@@ -471,14 +471,20 @@ end
 set -g ___fish_git_prompt_status_order stagedstate invalidstate dirtystate untrackedfiles
 
 function __fish_git_prompt_informative_status
-
     set -l changedFiles (command git diff --name-status | cut -c 1-2)
     set -l stagedFiles (command git diff --staged --name-status | cut -c 1-2)
+    set -l untrackedfiles 0
 
     set -l dirtystate (math (count $changedFiles) - (count (echo $changedFiles | grep "U")))
     set -l invalidstate (count (echo $stagedFiles | grep "U"))
     set -l stagedstate (math (count $stagedFiles) - $invalidstate)
-    set -l untrackedfiles (count (command git ls-files --others --exclude-standard))
+
+    # Untracked files may cause slow prompt generation.
+    # For this reason, it should be possible to avoid counting the files
+    # by setting __fish_git_prompt_showuntrackedfiles explicitly to "false".
+    if test "$__fish_git_prompt_showuntrackedfiles" != false
```

Comment:
If I would have to guess what the logic of an "informative" and "simple" mode should be, I'd say that they _should_ follow the same pattern.
However, the unconditional counting is the way it's done here, and I didn't want to make any changes to the logic or break backward compatibility.

I think that the entire thing is weird, and I am willing to rewrite it, but I made this change for this one specific use case I needed, and decided to make a pull request since it is useful, IMHO.